evaluation/evaluation_engine.py

```diff
@@ -226,33 +235,57 @@ def evaluate_text_summarization(
         f"The average bleu score on a subset of {dataset_name} = {predicted_summary_score}"
     )
     print(f"The average bleu score on its perturbed set = {transformed_summary_score}")
-
+    return {
+        "model_name": model_name,
+        "split": split,
+        "dataset_name": dataset_name,
+        "bleu": np.round(predicted_summary_score, 1),
+        "pt_bleu": np.round(predicted_summary_score, 1)
+    }
 
 def evaluate_text_classifier(
-    transformation, model_name, dataset_name, split="test[:20%]"
-):
+    transformation, model_name, dataset_name, split="test[:20%]", input_key=None):
+    def is_positive(label):
+        return label == 1 or (type(label) == str and "pos" in label.lower())
+    # TODO: extend the task to other classification tasks that's not sentiment analysis.
     # (1) load model
     if model_name is None:
         model_name = "aychang/roberta-base-imdb"
     # (2) load test set
     if dataset_name is None:
         dataset_name = "imdb"
-    print("Loading <%s> dataset to train <%s> model", dataset_name, model_name)
-    dataset = load_dataset(dataset_name, split=split)
+        input_key = "text"
+    print(f"Loading <{dataset_name}> dataset to evaluate <{model_name}> model.")
+    if dataset_name in ["qqp", "sst2"]:
```

Comment:
Probably QQP would need a separate interface than SST - taking in two sentences..